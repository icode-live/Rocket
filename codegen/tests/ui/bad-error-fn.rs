#![feature(plugin, decl_macro)]
#![plugin(rocket_codegen)]

extern crate rocket;

use rocket::{Error, Request};

#[catch(404)]
fn err_a(_a: Error, _b: Request, _c: Error) -> &'static str { "hi" }

#[catch(404)]
fn err_b(_a: (isize, usize)) -> &'static str { "hi" }
