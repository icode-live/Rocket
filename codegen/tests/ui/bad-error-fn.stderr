error: error catchers can have at most 2 arguments
 --> $DIR/bad-error-fn.rs:9:1
  |
9 | fn err_a(_a: Error, _b: Request, _c: Error) -> &'static str { "hi" }
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: error catchers can take either a `rocket::Error`, `rocket::Request` type, or both.

error: unknown error catcher argument
  --> $DIR/bad-error-fn.rs:12:14
   |
12 | fn err_b(_a: (isize, usize)) -> &'static str { "hi" }
   |              ^^^^^^^^^^^^^^
   |
   = help: error catchers can take either a `rocket::Error`, `rocket::Request` type, or both.

error: aborting due to 2 previous errors

